agnpy/spectra/tests/test_spectra.py

```diff
@@ -817,4 +824,107 @@ def test_compare_calculation_with_calculation_split_into_two(self):
             eval_2.evaluate(gammas, 1, gamma_min, gamma_max),
             0.001)
 
+    def test_total_number_of_electrons_should_stay_after_time_evolution(self):
+        r_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e48 * u.erg,
+            4 / 3 * np.pi * r_b ** 3,
+            p=2.8,
```

Comment:
it might be wise to use a harder spectral index (e.g. -1.5 instead of -2.8) for this test, because this will increase the number of electrons in the high edge of the spectrum that are most affected by the cooling

Reply:
I just tried with 1.5, but  it made the loss rate too high, and the time had to be split into too many subintervals. So i change it to 2.0 and reduced the total time.